core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -2792,27 +2792,28 @@ public static Collection multisetUnionAll(Collection collection1,
    * Function that, given a certain List containing single-item structs (i.e. arrays / lists with
    * a single item), builds an Enumerable that returns those single items inside the structs.
    */
-  public static Function1<Object, Enumerable<Comparable>> flatList() {
+  public static Function1<Object, Enumerable<Object>> flatList() {
```

Comment:
Are `null` values acceptable for the resulting function?
In other words, is `inputObject == null` valid or not?

If it is valid, then the signature should better be

```java
What do you think of the following signature?

```suggestion
  public static <T> Function1<T, Enumerable<T>> flatList() {
```

If null is not valid, then cast from `inputObject` to `List` should indeed be like `(List<Object>) ...` which effectively means `the list of non-nullable Object`

Reply:
{{SqlFunction#flatList}} was introduced in 1.24 via CALCITE-4063, and it is a counter-part of {{flatProduct}} for a very particular case. So, following {{flatProduct}}'s logic, inputObject cannot be null, so I guess {{flatList}} should look like:
```
  public static Function1<Object, Enumerable<Object>> flatList() {
    return inputObject -> {
      //noinspection unchecked
      final List<Object> list = (List<Object>) inputObject;
      final Enumerator<Object> enumerator = Linq4j.enumerator(list);
      ...
```
do you agree?

Reply:
Actually, I think we can do even better / simpler:
```
  public static Function1<List<Object>, Enumerable<Object>> flatList() {
    return inputObject -> {
      final Enumerator<Object> enumerator = Linq4j.enumerator(inputObject);
      ...
```

Reply:
Yes, that would be better indeed.
PS. `new Abstract ...` could probably be replaced with `Linq4j.asEnumerable(inputList).select(v -> ...)`

Reply:
Done. Definitely better, thanks for the tip!